ttnn/cpp/ttnn/operations/embedding_backward/device/kernels/compute/embedding_backward.cpp

```diff
@@ -0,0 +1,83 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "compute_kernel_api/eltwise_unary/eltwise_unary.h"
+#include "compute_kernel_api/tile_move_copy.h"
+#include "compute_kernel_api/reshuffle.h"
+#include "debug/dprint.h"
+
+auto sr = SliceRange{.h0 = 0, .h1 = 32, .hs = 8, .w0 = 0, .w1 = 32, .ws = 8};
+auto sr2 = SliceRange{.h0 = 31, .h1 = 32, .hs = 8, .w0 = 0, .w1 = 32, .ws = 8};
+
+//#define COMPUTE_DEBUG
+
+namespace NAMESPACE {
+void MAIN {
+    const uint32_t tiles_per_core = get_arg_val<uint32_t>(0);
+
+    constexpr uint32_t max_tiles_per_core = get_compile_time_arg_val(0);
+    constexpr uint32_t input_height = get_compile_time_arg_val(1);
+
+		constexpr uint32_t cb_grad = tt::CB::c_in0;
+		constexpr uint32_t cb_index = tt::CB::c_in1;
+		constexpr uint32_t cb_out_intermed = tt::CB::c_in2;
+		constexpr uint32_t cb_mask = tt::CB::c_intermed0;
+    constexpr uint32_t cb_chunk_count_scratch = tt::CB::c_intermed1;
+		constexpr uint32_t cb_out = tt::CB::c_out0;
+
+    unary_op_init_common(cb_grad);
+
+
+    for (uint32_t i = 0; i < input_height; ++i) {
+        cb_wait_front(cb_grad, max_tiles_per_core);
+
+        volatile uint32_t *chunk_addr_ptr;
+        cb_get_tile(cb_chunk_count_scratch, 0, &chunk_addr_ptr);
+
+        uint32_t chunk_count = chunk_addr_ptr[4]; // Need to shift because read ptr is off by 1 << 4 in BBE
```

Comment:
@TT-BrianLiu Could you explain in more detail why the pointer was shifted by that amount? (BBE stands for budabackend, right?)

Reply:
@rdjogoTT Can you comment? It's somewhere there is a decrement and then `<< 4` in BBE so the read ptrs of the CB are misaligned between NCRISC and UNPACK.